File: Qin/Qin-ios/Qin-ios/Util/Model/Group.swift
//
//  Group.swift
//  Qin-ios
//
//  Created by 杨璨 on 2020/2/14.
//  Copyright © 2020 canyang. All rights reserved.
//

import Foundation
import RealmSwift

public class Group {
    public var ID = ""
    public var Name = ""
    
    public static func GetGroup() {
        let checkJson = ["StudentID": ApplicationHelper.CurrentUser.ID]
        let checkJSONData = try? JSONSerialization.data(withJSONObject: checkJson, options: [])
        let jsonString = String(data: checkJSONData!, encoding: .utf8)
        let urlString = jsonString!.addingPercentEncoding(withAllowedCharacters: .urlHostAllowed)!
        let url = "\(DatabaseHelper.LeancloudAPIBaseURL)/1.1/classes/ReStudentGroup?where=\(urlString)"
        if let response = DatabaseHelper.LCSearch(searchURL: url) {
            let DatabaseResults = response["results"] as! [[String:Any?]]
            for checkLog in DatabaseResults {
                let groupID = checkLog["GroupID"] as! String
                ApplicationHelper.CurrentUser.GroupsID.append(groupID)
            }
        }
    }
}

File: Qin/Qin-Android/app/src/main/java/edu/bfa/ss/qin/Util/Group.java
package edu.bfa.ss.qin.Util;

import org.json.JSONArray;
import org.json.JSONObject;

import java.net.URLEncoder;

public class Group {
    public String ID = "";
    public String Name = "";

    public static void GetGroup() {
        try {
            String condition = URLEncoder.encode("{\"StudentID\":\"" + ApplicationHelper.CurrentUser.ID + "\"}", "UTF-8");
            String url = DatabaseHelper.LeancloudAPIBaseURL + "/1.1/classes/ReStudentGroup?where=" + condition;
            JSONObject response = DatabaseHelper.LCSearch(url);
            JSONArray DatabaseResults = response.getJSONArray("results");
            for (int i = 0; i < DatabaseResults.length(); i++) {
                JSONObject checkLog = DatabaseResults.getJSONObject(i);
                String groupID = checkLog.getString("GroupID");
                ApplicationHelper.CurrentUser.GroupsID.add(groupID);
            }
        } catch (Exception ignore) {
            return;
        }
    }
}
